torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -0,0 +1,1086 @@
+#include <torch/csrc/jit/tensorexpr/ir_simplifier.h>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+
+tensorexpr::SimplifierHashType Term::hashVars() const {
+  SimplifierHashType hash = 0;
+  for (auto* v : variables_) {
+    hash = hasher_.hash_combine(hash, hasher_.hash(v));
+  }
+
+  return hash;
+}
+
+void Term::sort() {
+  // order of ops important for float
+  if (dtype().is_floating_point()) {
+    return;
+  }
```

Comment:
I don't think we're preserving the order of floats since we're picking them from an `unordered_map` when putting them to `variables_`. Am I missing something?

Reply:
Floating point Exprs (even opaque ones) are never inserted into a Term or Polynomial, so we should be fine. I'll change this to throw an exception since it should be a logic error.